### DISCUSSION TITLE
v.3.0.0 - major update with many features and improvements

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#190] [Announcements]
🔥🔥🔥 New Year - New Version 🔥🔥🔥

## General
First of all, I hope you all had a great start into the new year!
How could we start the year better than with a big update :)

> ### I think it is clear to everyone that this project costs a lot of time and money.
> ### Therefore we came to the decision to introduce a donation link.
> ### Everyone can donate how much he/she wants to support this project and our efforts.
> ### So if you want to show your appreciation for our work, please follow the link below:

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

&nbsp;

## Files that need to be reloaded:

- nspanel_blueprint.yaml
- nspanel.tft
- nspanel_esphome.yaml

&nbsp;

### Community Updates
- sorted and fixed language selection - thanks @PunkRa1n
- added Fan entity to supported button entities - thanks @tikismoke
- added binary_sensor to supported button entities - thanks @naofireblade 

&nbsp;

### DEV Team Updates

- **Breaking change - new and easy way to select the weather entity**  🔥 **_!Attention - you have to set this after update to v.3.0.0 in your automation(s)_**
  This also fixes the problem with the default HA weather when HA is reinstalled
![Bildschirmfoto 2023-01-12 um 21 16 16](https://user-images.githubusercontent.com/41958506/212171610-02587e6f-7c67-429d-b5c9-91fc938090ea.png)

&nbsp;

- **Redesign of the home page**
  _With this we have found a new area for more buttons which enables the implementation of new functions. In this release, only the Notification button is currently being added. The next functions will be added step by step with the next releases._

_Here is a first glance on the new functions_

![Github-1-2](https://user-images.githubusercontent.com/41958506/212180508-f51e4842-78b3-4b92-95f1-f06c6aef9923.png)

&nbsp;

- **new function "add Indoor Temperature Sensor"**
  _It is now possible to add your own indoor temperature sensor to replace the internal NSPanel Temp Sensor on the home page_
![Bildschirmfoto 2023-01-12 um 21 48 34](https://user-images.githubusercontent.com/41958506/212178289-032ec758-81a9-42cb-993c-14cb8a4133ed.png)

&nbsp;

- **new function "Notification"** 
  _Now it is possible to send a notification to the NSPanel via HA. 
On the new notification page there is the possibility to mark the message as read and delete it or keep the message.
If the message is not deleted, a notification icon is displayed on the home page.
The notification icon can have two different colors - red -> unread and white -> read but not deleted._
![Bildschirmfoto 2023-01-12 um 21 45 33](https://user-images.githubusercontent.com/41958506/212177692-ca145ad3-b53c-4552-8634-4fa48546ffd4.png)

&nbsp;

- **new function "play a sound"**
  _It is now also possible to send a rtttl sound to the NSPanel via HA. This function can be useful for automations._

_Examples see here:_
https://esphome.io/components/rtttl.html#common-beeps 
or 
https://codebender.cc/sketch:109888#RTTTL%20Songs.ino

![Bildschirmfoto 2023-01-12 um 21 45 50](https://user-images.githubusercontent.com/41958506/212177541-206ed70d-b4e0-436c-9ef5-25e03a816898.png)

&nbsp;

- **new function "confirmation"**
  _It is now possible to enable confirmation via blueprint for each button. This means that if the button is pressed, this must first be   confirmed before the action is executed. Helpful for important things like front door, garden lighting or for children and older people._
![Bildschirmfoto 2023-01-12 um 21 51 46](https://user-images.githubusercontent.com/41958506/212178783-5035e065-daf5-404f-ba30-157873845a5b.png)

&nbsp;

- **added a button to disable the NSPanel sound** 
_added a button to ENABLE the NSPanel sound (default = off) for notification AND boot_ 
![Bildschirmfoto 2023-01-12 um 21 47 56](https://user-images.githubusercontent.com/41958506/212178213-19818fc0-c597-4998-8bc6-8ec0fb9ab811.png)

&nbsp;

- **add new HA Service for manual upload of a tft file** 
  _(for example the blank_tft can be uploaded if there are problems with the TFT file)_
![Bildschirmfoto 2023-01-12 um 21 12 39](https://user-images.githubusercontent.com/41958506/212170902-a0747171-9849-4955-ab51-2ada394828d8.png)

&nbsp;

- **new boot menu with reboot button.** 
  _It will now only jump to the home page when the boot process has been completed successfully. Reboot button on the boot page only required if boot does not succseed_
![Github-1-3](https://user-images.githubusercontent.com/41958506/212180568-3e48217b-137d-4ff7-b875-12fcc8ced936.png)

&nbsp;

- Some changes in the code for better performance and small bugfixes

&nbsp;

## Next topics we are currently working on
See here: https://github.com/Blackymas/NSPanel_HA_Blueprint/labels/roadmap

<hr /><em>This discussion was created from the release <a href='https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v.3.0.0'>v.3.0.0 - major update with many features and improvements</a>.</em>

**Comment by dirk898** [comment]
Hi 
i updated to version 3.0 and added an indoor temperatur sensor. Now the display jumps between the temperatur of the newly added sensor and the one of the panel itself. I tryied deactivating the panel sensor but that didn´t work.

Also i get an error message in home assistant: nspanel calls unknown service: "esphome.nspanelbad_notification_clear".

Any ideas why?

Thanks for the amazing blueprint it was a game changer for me.

**Reply by Blackymas** [reply]
have you updated all 3 Files Blueprint, Esphome and TFT?

**Reply by dirk898** [reply]
can you explain how to update the esphome.yaml?
do i copy the esphome.yaml from your github and replace it with the one in my /config/esphome/nspanelbad.yaml ?

thanks for the reply

**Reply by Blackymas** [reply]
here the steps

1. update ESPhome -> validate -> install
2. HA -> Device -> TFT Upload
3. copy nspanel_blueprint.yaml -> add in the right document -> restart automations
4. automations -> NSPanel -> language and weather selected again -> automation saved
5. reboot NSPanel


https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/183

**Reply by dirk898** [reply]
works like a charm

i just had to update within esphome in sidebar, sorry for the inconvenience
